scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/json/SchedulerJSONUtils.java

```diff
@@ -274,4 +266,20 @@ public static String getStringOrElse(JSONObject jsonObject, String attributeName
         }
         return Collections.emptyMap();
     }
+
+    public static Set<String> extractSet(JSONValue setValue) {
+        if (setValue != null) {
+            JSONArray keyValueArray = setValue.isArray();
+            if (keyValueArray != null) {
+                int arraySize = keyValueArray.size();
+                Set<String> resultSet = new HashSet<>(arraySize);
+                for (int i = 0; i < keyValueArray.size(); i++) {
+                    JSONString jsonString = keyValueArray.get(i).isString();
+                    resultSet.add(jsonString.stringValue());
+                }
+                return resultSet;
+            }
+        }
+        return Collections.emptySet();
+    }
```

Comment:
In my opinion Java8 style is simpler to read. 

For instance:

```
    public static Set<String> extractSet(JSONValue setValue) {
        if (setValue == null || setValue.isArray() == null) {
            return Collections.emptySet();
        } else {
            JSONArray keyValueArray = setValue.isArray();
            return IntStream.range(0, keyValueArray.size())
                        .mapToObj(keyValueArray::get)
                        .map(JSONValue::isString)
                        .map(JSONString::stringValue)
                        .collect(Collectors.toSet());
        }
    }
```

Your call